### PR TITLE
Remove autotools

### PR DESCRIPTION
**Pull request by notspiff**
This is obviously ahead of its time, but for tracking purposes and for starting to root out possible issues, I have created this PR.

One related question is whether or not you want to relocate the cmake buildsystem to the top of the tree.

### CODE REVIEW

**Comment by akva2** [comment]
can we somehow run a depends based platform using a cmake build on jenkins yet ? curious if there are more shared things between the configures i didn't catch..

**Comment by fritsch** [comment]
We can. I was told cmake builds are the defaults now, so let's try it:

jenkins build this please

**Comment by hudokkow** [comment]
CMake relocation is planned. Are you volunteering? :wink:

**Comment by akva2** [comment]
sure. if there is some time still, i dont have many hours to put in these days.

**Comment by fetzerch** [comment]
Good idea to test that already, thx :)

I'd also recommend to move CMake to the root tree. There are a few IDEs (KDevelop, QtCreator) that have some issues if the main CMakeLists.txt is in a subdirectory and it'll be just easier for everyone.  

For the current state:
- Android should be fine now
- For rbpi, this lines have to be removed: https://github.com/xbmc/xbmc/blob/master/tools/buildsteps/rbpi/prepare-xbmc#L6

Few more things to do, I just want to mention:
- All conditional CMAKE_BUILD stuff in tools/buildsteps can go
- Drop Kodi.xcodeproj
- docs/Readme.xyz need to be adapted

**Comment by hudokkow** [comment]
Heads up: https://github.com/hudokkow/xbmc/tree/cmake_move_to_root

**Comment by hudokkow** [comment]
Can you rebase please?

**Comment by notspiff** [comment]
done.

**Comment by hudokkow** [comment]
thx!
jenkins build this with addons please

**Comment by hudokkow** [comment]
Only RBPi failed: http://jenkins.kodi.tv/job/LINUX-RBPI/8630/console
> ++ JSON_BUILDER=/home/jenkins/workspace/LINUX-RBPI/tools/depends/xbmc-depends/i686-linux-gnu-native/bin/JsonSchemaBuilder
++ ./bootstrap
/home/jenkins/workspace/LINUX-RBPI/tools/buildsteps/rbpi/prepare-xbmc: line 6: ./bootstrap: No such file or directory
Build step 'Execute shell' marked build as failure
Set build name.
New build name is 'xbmc-PR10429-1ec7b39'
Variable with name 'BUILD_DISPLAY_NAME' already exists, current value: 'xbmc-PR10429-1ec7b39', new value: 'xbmc-PR10429-1ec7b39'
Finished: FAILURE

http://jenkins.kodi.tv/view/Automation/job/BuildMultiWithAddons-PR/48/

**Comment by akva2** [comment]
well rbp still uses the autotools build so that is to be expected.

**Comment by fetzerch** [comment]
Nope rbpi is also built with cmake, but:

> For rbpi, this lines has to be removed: https://github.com/xbmc/xbmc/blob/master/tools/buildsteps/rbpi/prepare-xbmc#L6

**Comment by akva2** [comment]
thanks, removed.

**Comment by fetzerch** [comment]
hudo was faster kicking off the build ;) Builds fine now also on rbpi: http://jenkins.kodi.tv/job/LINUX-RBPI/8650/

**Comment by hudokkow** [comment]
:stuck_out_tongue_winking_eye: 

Arne, unless we decide to generate a config.h file with CMake, you might want to pick and squash https://github.com/hudokkow/xbmc/commit/faa43f2cc915bb89cc843c573e45a5bc8b95cbfa

There's probably other defines in need of cleanup but it's hard to find them.

**Comment by akva2** [comment]
pulled. there was another commit in the branch but i left it out.

**Comment by hudokkow** [comment]
Thanks. The other commit isn't related. It was just to make sure I didn't killed win builds after that ifdeferry cleanup.

**Comment by stefansaraev** [comment]
it's time..

**Comment by fetzerch** [comment]
I'll merge https://github.com/xbmc/xbmc/pull/10509 now if it still builds fine.
We should however get the confirmation from @Memphiz that cmake works for him as the legacy xcode build depends on autotools.

**Comment by akva2** [comment]
rebased

**Comment by hudokkow** [comment]
:+1: 
jenkins build this with addons please

**Comment by hudokkow** [comment]
All droids failed on the same spot:

>CMake Error: File /home/jenkins/workspace/Android-X86/tools/android/packaging/Makefile.in does not exist.
CMake Error at scripts/android/Install.cmake:17 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:426 (include)

http://jenkins.kodi.tv/job/Android-X86/9186/console
http://jenkins.kodi.tv/job/Android-ARM/10298/consoleFull
http://jenkins.kodi.tv/job/Android-ARM64-v8a/795/consoleFull

**Comment by akva2** [comment]
sorry i nuked every file i could find ;) didn't realize droid packaging used that file still. it's back.

**Comment by hudokkow** [comment]
heheh, np.
jenkins build this with addons please

File: tools/android/packaging/Makefile.in [hunk 1]
```diff
@@ -1,6 +1,7 @@
 include ../../depends/Makefile.include
 
 OBJS = libcurl.so \
+  librtmp.so \
```
Comment (hudokkow) [inline tools/android/packaging/Makefile.in]: > cp: cannot stat '/home/jenkins/workspace/Android-ARM/build/install/lib/librtmp.so': No such file or directory
Makefile:130: recipe for target 'libs' failed
make[4]: *** [libs] Error 1
CMakeFiles/apk-unsigned.dir/build.make:57: recipe for target 'CMakeFiles/apk-unsigned' failed
make[3]: *** [CMakeFiles/apk-unsigned] Error 2
CMakeFiles/Makefile2:1202: recipe for target 'CMakeFiles/apk-unsigned.dir/all' failed
make[2]: *** [CMakeFiles/apk-unsigned.dir/all] Error 2
CMakeFiles/Makefile2:1209: recipe for target 'CMakeFiles/apk-unsigned.dir/rule' failed
make[1]: *** [CMakeFiles/apk-unsigned.dir/rule] Error 2
Makefile:378: recipe for target 'apk-unsigned' failed
make: *** [apk-unsigned] Error 2
Build step 'Execute shell' marked build as failure

**Comment by notspiff** [comment]
doh! i probably copied an old file. !"#¤!"#¤!"#¤!

**Comment by hudokkow** [comment]
No need to apologize. Thx.
jenkins build this with addons please and behave this time.

**Comment by hudokkow** [comment]
Good news is this now [builds](http://jenkins.kodi.tv/view/Automation/job/BuildMultiWithAddons-PR/70/).
One of the droids had a brain fart but that's normal.

Bad news is it needs a(nother) rebase. Since @AlwinEsch is touching a lot of the same files with his add-ons cleanup/refactor, may I suggest that we let him finish it, rebase, test again and get it in?

@AlwinEsch, can you give us an heads up when you're finished? Is https://github.com/xbmc/xbmc/pull/11172 the last of its kind?
@wsnipex, you OK with this going in?

**Comment by wsnipex** [comment]
fine for me

**Comment by hudokkow** [comment]
Can you rebase please? Thx

**Comment by notspiff** [comment]
done.

**Comment by hudokkow** [comment]
jenkins build and merge

**Comment by MartijnKaijser** [comment]
Failure on android build

**Comment by hudokkow** [comment]
All good: http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1838/

Goodbye old friend.
Thx Arne!

**Comment by notspiff** [comment]
6 years. a shitload of pain. let me take the opportunity to say 'HAHA' to
those that said it couldn't be done. to everybody else: thank you from the
bottom of my heart for helping out.

**Comment by hudokkow** [comment]
Congrats!

**Comment by MartijnKaijser** [comment]
@notspiff thanks a lot :)

**Comment by hauihau** [comment]
Great job guys! :)

**Comment by wsnipex** [comment]
yeah, great job!

**Comment by fetzerch** [comment]
omg it happened! Thanks all :)

**Comment by garbear** [comment]
hm, this would useful in my retroplayer branch. feel like backporting? ;)

**Comment by ksooo** [comment]
Hi, afaik building (dependencies) on macOS relies on autotools. How do I build stuff on macOS now?

**Comment by stefansaraev** [comment]
just like before. cd tools/depends;./configure -- .....

**Comment by Rechi** [comment]
I think https://github.com/xbmc/xbmc/tree/dea9386/tools/depends/target/xbmc is useless now.

**Comment by ksooo** [comment]
macOS: works just fine. nice. thanks

**Comment by Memphiz** [comment]
@fetzerch care to update docs/README.osx and docs/README.ios with the newest build instructions?

**Comment by fetzerch** [comment]
> @fetzerch care to update docs/README.osx and docs/README.ios with the newest build instructions?

Shit, https://github.com/xbmc/xbmc/blob/master/docs/README.osx#L125 now just reminded me that I have forgotten about a small but very important detail. osx/ios builds still need `make xcode_depends` which builds some libs like cpluff, libexif and the addon callback libs. Really sorry bout that :(

I'd see the following choices to go on:
- xcode users switch to the cmake generated xcode project (and I help ironing out the issues you run into)
- we get the removed functionality back (maybe @wsnipex could check how much that is for the makefiles), in parallel I can look into what would be needed to build only those libs with cmake.
- we fully revert the change until you're happy with the generated xcode project and xcode users switched to cmake (I don't think that's a good solution considering the amount of work spent rebasing this change over and over again, especially now that we have pretty invasive changes going into master)

**Comment by hudokkow** [comment]
No revert, please. Last resource solution, imo.

**Comment by Memphiz** [comment]
First option it is. Then this needs to be described in the readm

**Comment by FernetMenta** [comment]
what is the current state? as far as I understand it building for OSX is broken because things like libdvd are not built?

**Comment by Memphiz** [comment]
As far as understand it's not as long as you use cmake - the Xcode project is useless now though

**Comment by FernetMenta** [comment]
ah ok. why don't we drop the old xcode project?

**Comment by fetzerch** [comment]
I'll update the readme and explain how to build with cmake.
For the moment: https://github.com/xbmc/xbmc/blob/master/project/cmake/README.md

**Comment by hauihau** [comment]
cmake in contrast to autotools now also installs /usr/bin/TexturePacker. Correct me if I'm wrong, but texturepacker is only used while building. Wouldn't it make sense to skip it from installing?

**Comment by hudokkow** [comment]
^^ Something to discuss with @cjmayo for https://github.com/xbmc/xbmc/pull/11213?

**Pull request merged**